### DISCUSSION TITLE
Validating classification (and other) payloads

### DISCUSSION THREAD

**Discussion by rogerhutchings** [zooniverse/front-end-monorepo#2447] [Ideas]
- What about using something like [Joi](https://github.com/hapijs/joi) to validate data before we submit it to the API?
- We could define our schemas as something implementation-agnostic (possibly [JSON Schema](https://json-schema.org/)?) for external use, and write a converter for Joi?
- The validation step could be done seamlessly by implementing it as a middleware for Superagent / Frisbee / whatever

cc'ing @eatyourgreens because we were talking about it earlier

**Comment by srallen** [comment]
This seems related to this proposed ADR: https://github.com/zooniverse/front-end-monorepo/blob/master/docs/arch/adr-7.md
